test/ressources/utils/matomo_tracker_setup.dart

```diff
@@ -4,16 +4,15 @@ import 'package:mocktail/mocktail.dart';
 import '../mock/data.dart';
 import '../mock/mock.dart';
 
-void matomoTrackerSetup() {
+void matomoTrackerSetup({String? visitorId}) {
   TestWidgetsFlutterBinding.ensureInitialized();
 
-  when(mockLocalStorage.getVisitorId).thenAnswer((_) async => null);
+  when(mockLocalStorage.getVisitorId).thenAnswer((_) async => visitorId);
```

Comment:
No need to pass the `visitorId` to the mock, you can already override the stub for `getVisitorId` by doing another `when()` in your test.